resolver_limited.go

```diff
@@ -91,12 +121,19 @@ func (r *LimitedResolver) MatchMX(name string, matcher IPMatcherFunc) (bool, *Re
 	}
 
 	limit := int32(r.mxQueriesLimit)
-	return r.resolver.MatchMX(name, func(ip net.IP, name string) (bool, error) {
+	found, extras, err := r.resolver.MatchMX(name, func(ip net.IP, name string) (bool, error) {
 		if atomic.AddInt32(&limit, -1) < 1 {
 			return false, ErrDNSLimitExceeded
 		}
 		return matcher(ip, name)
 	})
+	if extras != nil && extras.Void {
+		if !r.canPerformVoidLookup() {
```

Comment:
Did you consider adding ```func isVoid(v *ResponseExtrase) bool { return v!=nil && v.Void }```?

then this bit could be simplified to ```if isVoid(extras) && !r.canPerformVoidLookup() { ... }```

Reply:
done